src/main/java/io/github/thebusybiscuit/exoticgarden/PlantsListener.java

```diff
@@ -115,75 +176,70 @@ public void onGrow(StructureGrowEvent e) {
         }
     }
 
-    @EventHandler
-    public void onGenerate(ChunkPopulateEvent e) {
-        if (!SlimefunPlugin.getWorldSettingsService().isWorldEnabled(e.getWorld())) {
-            return;
+    private void pasteTree(ChunkPopulateEvent e, int x, int z, Tree tree) {
+        for (int y = e.getWorld().getMaxHeight(); y > 30; y--) {
+            Block current = e.getWorld().getBlockAt(x, y, z);
+            if (!current.getType().isSolid() && current.getType() != Material.WATER && current.getType() != Material.SEAGRASS && current.getType() != Material.TALL_SEAGRASS && !(current.getBlockData() instanceof Waterlogged && ((Waterlogged) current.getBlockData()).isWaterlogged()) && tree.isSoil(current.getRelative(0, -1, 0).getType()) && isFlat(current)) {
+                Schematic.pasteSchematic(new Location(e.getWorld(), x, y, z), tree);
+                break;
+            }
         }
+    }
 
-        if (!cfg.getStringList("world-blacklist").contains(e.getWorld().getName())) {
-            Random random = ThreadLocalRandom.current();
-
-            if (random.nextInt(100) < cfg.getInt("chances.BUSH")) {
-                Berry berry = ExoticGarden.getBerries().get(random.nextInt(ExoticGarden.getBerries().size()));
-                if (berry.getType().equals(PlantType.ORE_PLANT)) return;
-
-                int x = e.getChunk().getX() * 16 + random.nextInt(16);
-                int z = e.getChunk().getZ() * 16 + random.nextInt(16);
-
-                for (int y = e.getWorld().getMaxHeight(); y > 30; y--) {
-                    Block current = e.getWorld().getBlockAt(x, y, z);
-                    if (!current.getType().isSolid() && current.getType() != Material.WATER && berry.isSoil(current.getRelative(BlockFace.DOWN).getType())) {
-                        BlockStorage.store(current, berry.getItem());
-                        switch (berry.getType()) {
-                        case BUSH:
+    private void blockSet(ChunkPopulateEvent e, int x, int z, Berry berry, Random random, boolean isPaper) {
```

Comment:
This method name is a little off too. Since this method is solely for "Berries" the name "blockSet" seems rather generic.
Something like `growBush` or so sounds better.